### ISSUE TITLE
Phase out the use of `x-requested-with` and replace with discrete URLs

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#9272]
> **Important:** This is a longer term migration plan, new code should use the new conventions but old code can probably be left as is unless other changes are being made.

### Is your proposal related to a problem?

- Wagtail is slowly moving away from jQuery and some of the conventions and approaches that are part of the jQuery core code, specifically the reliance on `x-requested-with` header to determine if the request has been made async.
- Additionally, Django is moving away from this header usage in its core code and documentation.

### Describe the solution you'd like

- Migrate towards discrete URLs, with discrete names/views to provide async results for listings (for example, search results being patched into the DOM as you type).
- Migrate away from the reliance of `x-requested-with` for pure JSON results (e.g. page explorer and similar), instead using the JSON accepts header is sufficient.

#### Known usage 

Note. There may be others, this doesn't cover code that adds this header to requests. 

https://github.com/wagtail/wagtail/blob/23275a4cef4d36bb311abe315eb9ddfc43868e8b/wagtail/admin/views/generic/multiple_upload.py#L41

https://github.com/wagtail/wagtail/blob/23275a4cef4d36bb311abe315eb9ddfc43868e8b/wagtail/admin/views/generic/workflow.py#L135

### Describe alternatives you've considered

- There may be security reasons to keep some kind of header https://stackoverflow.com/a/22533680
- It may be simpler not to change anything and instead let the code live as is.

### Additional context

- `x-requested-with` us added via jquery.ajax by default https://api.jquery.com/jquery.ajax/
- Django 3.1 removed the `request.is_ajax()` method that returned true if the `x-requested-with` header existed - https://docs.djangoproject.com/en/4.1/releases/3.1/#id2
- In other places, such as the choosers, we've phased out the use of `x-requested-with` in favour of having separate views at distinct URLs such as `/admin/users/` and `/admin/users/results/`, for the full page response versus the HTML fragment. (Of course, class-based views make this really easy, since they can share the same base class and just override the `template_name` property.) _Originally posted by @gasman in https://github.com/wagtail/wagtail/pull/9114#discussion_r978122987_

**Comment by yokwejuste** [comment]
I guess this issue should be solved by now! Is this issue still to be @lb-

**Comment by lb-** [comment]
@yokwejuste yes this is still an issue, it is low priority as nothing is breaking with the current approach.

Feel free to pick up this but just try to change ONE location first in a single PR, see the discussion linked above for more context.

Reach out if you need a hand, but this will require some solid knowledge of how Django's URLs system works, how async JS requests work, and how dynamic search results work in the admin. May not be a good first issue sorry.

**Comment by yokwejuste** [comment]
Thanks @lb- I'm on it!

In case I need help I'll get to you.

**Comment by yokwejuste** [comment]
Hi 

Is this still of actuality? Will like to Work on

**Comment by lb-** [comment]
@yokwejuste my comment above still applies https://github.com/wagtail/wagtail/issues/9272#issuecomment-1276767409

**Comment by yokwejuste** [comment]
great, working on it then!

**Comment by dhruvipatel14** [comment]
Hi @lb- ,

I’ve reviewed the various uses of `x-requested-with` in the codebase, and here are my findings:

1. Usage in `auth.py`: 
It is used to raise a Permission Denied error, seemingly to maintain security.
https://github.com/wagtail/wagtail/blob/784f4bbbb920c21895292e50d66659038f1acf46/wagtail/admin/auth.py#L18-L21

2. In `MultipleUploads`:
When any document or image gets updated, `x-requested-with` is added via `jquery.ajax`.
https://github.com/wagtail/wagtail/blob/784f4bbbb920c21895292e50d66659038f1acf46/wagtail/admin/views/generic/multiple_upload.py#L41-L46

3. Usage in `swapController`:
As I’m new to the project, I’m not entirely sure of its use case here. It appears to be part of a core controller in the project.
https://github.com/wagtail/wagtail/blob/784f4bbbb920c21895292e50d66659038f1acf46/client/src/controllers/SwapController.ts#L294-L305

4. Usage in Workflow:
It is used to raise errors if a form is not valid. I’ve reviewed the tests but am unsure how to reproduce this issue in the admin. One suggestion is to use a separate template for rendering form errors, similar to how search results are rendered in the chooser.
https://github.com/wagtail/wagtail/blob/784f4bbbb920c21895292e50d66659038f1acf46/wagtail/admin/views/generic/workflow.py#L133-L152

I’d appreciate your guidance on how to approach this. Let me know how I should proceed, particularly with reproducing the workflow error scenario in the admin to check the use case of using `x-requested-with`.
